migrations/1641994076462-fix_ldap_federal_state.js

```diff
@@ -0,0 +1,84 @@
+const mongoose = require('mongoose');
+const { ObjectId } = require('mongoose').Types;
+const { alert, error } = require('../src/logger');
+const { connect, close } = require('../src/utils/database');
+const federalStateModel = require('../src/services/federalState/model');
+const systemModel = require('../src/services/system/model');
+const { schoolSchema } = require('../src/services/school/model');
+
+const schoolModel = mongoose.model('schools_20211021', schoolSchema, 'schools');
+
+const getFederalStateFromSchool = async (systemId) => {
+	// is this ldap configured to more than 1 school?
+	const federalStates = await schoolModel
+		.aggregate([{ $match: { systems: { $in: [ObjectId(systemId)] } } }, { $group: { _id: '$federalState' } }])
+		.exec();
+
+	if (federalStates.length === 0) {
+		error(`LDAP system ${systemId} is not associated to any school`);
+		return;
+	}
+	if (federalStates.length > 1) {
+		error(`LDAP system ${systemId} used in multiple schools from different federal states`);
+		return;
+	}
+	if (federalStates[0]._id === null) {
+		error(`LDAP system ${systemId} cannot be associated with a federal state`);
+		return;
+	}
+	return federalStates[0]._id;
+};
+
+const getFederalStates = async (federalStates) =>
+	federalStateModel.findOne({ abbreviation: federalStates }).select('_id').lean().exec();
+
+module.exports = {
+	up: async function up() {
+		await connect();
+
+		const { _id: brandenburg } = await getFederalStates('BB');
+		const { _id: lowersaxony } = await getFederalStates('NI');
+
+		const systemsData = await systemModel
+			.find({
+				type: { $in: ['ldap', 'iserv'] },
```

Comment:
The system type "iserv" so not the same as the ldap provider is set to iserv.
Systems from the type iserv are SSO Systems and systems from type ldap are use the ldap provider to determin there working.
```suggestion
				type: 'ldap',
```